### PR TITLE
[12.0] Support for Variants, Fix for issue: No Barcode Generation on variants #260

### PR DESCRIPTION
**Pull request by heymanh**
Fix for issue [12.0] No Barcode Generation on variants #260
Adds barcode generation on variants
Removes barcode generation on initial product if product has variants

### CODE REVIEW

File: barcodes_generator_product/views/view_product_product.xml [hunk 1]
```diff
@@ -11,6 +11,44 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
...
+                <field name="generate_type" invisible="1" />
+                <field name="barcode_base" attrs="{
+                    'invisible': ['|',('barcode_rule_id', '=', False),('product_variant_count', '&gt;', 1)],
+                    'readonly': [('generate_type', '!=', 'manual')]}"
+                    groups="barcodes_generator_abstract.generate_barcode" colspan="2"/>
+                <button name="generate_base" type="object" string="Generate Base (Using Sequence)"
+                        attrs="{'invisible': ['|','|',
+                    ('generate_type', '!=', 'sequence'),
+                    ('barcode_base', '!=', 0),
+                    ('product_variant_count', '&gt;', 1)
```
Comment (chienandalu) [inline barcodes_generator_product/views/view_product_product.xml]: Why to hide it in this case? Isn't it the opposite to what you try to achieve?

**Review by chienandalu — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution :)

I don't understand well the reason why you hide de barcode controls in the regular variants form and show it only in `product_variant_easy_edit_view`.

File: barcodes_generator_product/views/view_product_product.xml (hunk 1 above)
Comment (heymanh) [reply barcodes_generator_product/views/view_product_product.xml]: Hi @chienandalu 
You are very welcome ;)
Actually it's not in the variants form but in the product_normal_form_view that we hide the barcode if variants exist.
Regards

Comment (chienandalu) [reply barcodes_generator_product/views/view_product_product.xml]: Well, that's not true when you go to the variant from Catalog > Product Variants :)

Comment (heymanh) [reply barcodes_generator_product/views/view_product_product.xml]: It's another view?!
I can't find the menu catalog > Products Variants !

Comment (chienandalu) [reply barcodes_generator_product/views/view_product_product.xml]: Sorry. Master Data > Product Variants :)

Comment (heymanh) [reply barcodes_generator_product/views/view_product_product.xml]: Sorry but I can't find anything like "master data"
It's a menu right? From witch app do I access it?

Comment (chienandalu) [reply barcodes_generator_product/views/view_product_product.xml]: Inventory > Master Data > Product Variants

You can test in this PR runbot

Comment (heymanh) [reply barcodes_generator_product/views/view_product_product.xml]: I understand now what you meant from the beginning :)
I've committed a fix
thank you for your support

**Pull request closed without merging**